### ISSUE TITLE
pip install pyodbc failing: error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

### ISSUE THREAD

**Issue by wondernath** [mkleehammer/pyodbc#276]
Please first make sure you have looked at:

* Documentation: https://github.com/mkleehammer/pyodbc/wiki
* Other issues

### Environment

To diagnose, we usually need to know the following, including version numbers.  On Windows, be
sure to specify 32-bit Python or 64-bit:

- Python: 2.7.6
- pyodbc: Latest version
- OS: ubuntu 14.04.5
- DB:
- driver:

### Issue
pip install pyodbc --user
Collecting pyodbc
  Using cached pyodbc-4.0.17.tar.gz
Building wheels for collected packages: pyodbc
  Running setup.py bdist_wheel for pyodbc ... error
  Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-YtLrrj/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp5eqhA5pip-wheel- --python-tag cp27:
  running bdist_wheel
  running build
  running build_ext
  building 'pyodbc' extension
  creating build
  creating build/temp.linux-x86_64-2.7
  creating build/temp.linux-x86_64-2.7/src
  x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=4.0.17 -DSQL_WCHART_CONVERT=1 -I/usr/include/python2.7 -c src/textenc.cpp -o build/temp.linux-x86_64-2.7/src/textenc.o -Wno-write-strings
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
  In file included from src/textenc.cpp:2:0:
  src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
   #include <sql.h>
                   ^
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

  ----------------------------------------
  Failed building wheel for pyodbc
  Running setup.py clean for pyodbc
Failed to build pyodbc
Installing collected packages: pyodbc
  Running setup.py install for pyodbc ... error
    Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-YtLrrj/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-CRLrEJ-record/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    running install
    running build
    running build_ext
    building 'pyodbc' extension
    creating build
    creating build/temp.linux-x86_64-2.7
    creating build/temp.linux-x86_64-2.7/src
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=4.0.17 -DSQL_WCHART_CONVERT=1 -I/usr/include/python2.7 -c src/textenc.cpp -o build/temp.linux-x86_64-2.7/src/textenc.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
    In file included from src/textenc.cpp:2:0:
    src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
     #include <sql.h>
                     ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-YtLrrj/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-CRLrEJ-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-YtLrrj/pyodbc/

**Comment by gordthompson** [comment]
> src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
> #include <sql.h>
> ^

Your machine does not have the ODBC header files installed. You need to do

```
sudo apt-get install unixodbc-dev
```

and then try installing pyodbc again.

**Comment by fenchu** [comment]
https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server

**Comment by kabirbaidhya** [comment]
Hi,

I already have `unixodbc-dev` installed on my system. I followed [this](https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server#microsoft-odbc-driver-17-for-sql-server).

```
➜ sudo apt-get install unixodbc-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
unixodbc-dev is already the newest version (2.3.1-4.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

And I'm trying to install the latest `pyodbc==4.0.22`; still getting the same error. Any idea on this?
```
➜ pip install pyodbc==4.0.22                                                                                                      
Collecting pyodbc==4.0.22
  Using cached pyodbc-4.0.22.tar.gz
Building wheels for collected packages: pyodbc
  Running setup.py bdist_wheel for pyodbc ... error
  Complete output from command /home/kabir/envs/etl/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-kp3jjwpo/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpqpqsm7h7pip-wheel- --python-tag cp35:
  running bdist_wheel
  running build
  running build_ext
  building 'pyodbc' extension
  creating build
  creating build/temp.linux-x86_64-3.5
  creating build/temp.linux-x86_64-3.5/src
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DPYODBC_VERSION=4.0.22 -I/usr/include/python3.5m -I/home/kabir/envs/etl/include/python3.5m -c src/buffer.cpp -o build/temp.linux-x86_64-3.5/src/buffer.o -Wno-write-strings
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  In file included from src/buffer.cpp:12:0:
  src/pyodbc.h:45:20: fatal error: Python.h: No such file or directory
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for pyodbc
  Running setup.py clean for pyodbc
Failed to build pyodbc
Installing collected packages: pyodbc
  Running setup.py install for pyodbc ... error
    Complete output from command /home/kabir/envs/etl/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-kp3jjwpo/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-4setzddr-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/kabir/envs/etl/include/site/python3.5/pyodbc:
    running install
    running build
    running build_ext
    building 'pyodbc' extension
    creating build
    creating build/temp.linux-x86_64-3.5
    creating build/temp.linux-x86_64-3.5/src
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DPYODBC_VERSION=4.0.22 -I/usr/include/python3.5m -I/home/kabir/envs/etl/include/python3.5m -c src/buffer.cpp -o build/temp.linux-x86_64-3.5/src/buffer.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    In file included from src/buffer.cpp:12:0:
    src/pyodbc.h:45:20: fatal error: Python.h: No such file or directory
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/home/kabir/envs/etl/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-kp3jjwpo/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-4setzddr-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/kabir/envs/etl/include/site/python3.5/pyodbc" failed with error code 1 in /tmp/pip-build-kp3jjwpo/pyodbc/
```


I'm using:
Python 3.5.2
Ms SQL Server 2017
Ubuntu 16.04 LTS

**Comment by kabirbaidhya** [comment]
Anyway, I figured out the issue. I had to install `python3-dev` on ubuntu and it did work.

```
➜ sudo apt-get install python3-dev
```

FYI : Just in case anyone runs into this issue, python 2.x users, will need `python-dev` instead.

**Comment by neostorm** [comment]
I'm see the same issue and both python3-dev and unixodbc-dev are installed.
I've looked elsewhere and installed every recommended package any other thoughts on how fix this?

Python 3.6.4
Ubuntu 16.04.2 LTS
```    
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    In file included from src/row.cpp:12:0:
    src/pyodbc.h:45:20: fatal error: Python.h: No such file or directory
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```

**Comment by kabirbaidhya** [comment]
Strange. I exactly had same issue.
```
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DPYODBC_VERSION=4.0.22 -I/usr/include/python3.5m -I/home/kabir/envs/etl/include/python3.5m -c src/buffer.cpp -o build/temp.linux-x86_64-3.5/src/buffer.o -Wno-write-strings
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    In file included from src/buffer.cpp:12:0:
    src/pyodbc.h:45:20: fatal error: Python.h: No such file or directory
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```

But after installing `python3-dev` and `unixodbc-dev`, it got resolved.

Can you make sure you have all these dev packages installed on your system? Not sure if this works, but these include pretty much all the dependencies usually required for python.
```
➜ sudo apt-get install build-essential libssl-dev libffi-dev python3-dev
➜ sudo apt-get install unixodbc-dev
```

**Comment by gordthompson** [comment]
@neostorm - How did you install Python 3.6 on that machine?

**Comment by carlosmarin** [comment]
RHEL / CentOS:
```
yum install unixODBC
yum install unixODBC-devel
```

**Comment by vibhusheet** [comment]
I have a ubuntu with all of the dependencies installed, the python 2x and 3x versions installed directly on the machine are able to install pyodbc
But, I encounter the same issue when I create a virtualenv for python3.6 and install pyodbc 

there it gives me:

```
    building 'pyodbc' extension
    creating build
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/src
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DPYODBC_VERSION=4.0.23 -I/usr/include/python3.6m -I/home/vibhu/workspace/test/PL_EMI_Reports/env/include/python3.6m -c src/getdata.cpp -o build/temp.linux-x86_64-3.6/src/getdata.o -Wno-write-strings
    cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
    In file included from src/getdata.cpp:5:0:
    src/pyodbc.h:45:20: fatal error: Python.h: No such file or directory
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1 ```

How do I fix this ?

**Comment by ikwijaya** [comment]
using alpine??

**Comment by ffedoroff** [comment]
for python3.7 run also:
`sudo apt-get install python3.7-dev`

**Comment by luissamano** [comment]
> for python3.7 run also:
> `sudo apt-get install python3.7-dev`

Thanks, It worked.

**Comment by dxkaufman** [comment]
I have the same issue in Visual Studio Code on Mac OSX while trying to deploy a timer trigger function to Azure Functions.  The function works perfectly while executing within VSC, but the deployment fails with the same 'unable to execute gcc' error.  I found another issue talking about that, where I've asked for help too, but if anyone watching this issue can help me, I'd be very grateful!

**Comment by v-chojas** [comment]
@dxkaufman what you're describing has nothing to do with pyODBC, and that error message is very generic, so it is no surprise you will find issues about it which are completely unrelated to your situation.

**Comment by dxkaufman** [comment]
@v-chojas For that matter, I seem to have read this post inattentively... I thought it mentioned the gcc error, but now I realize this one has more to do with the presence of code files... my mistake.  Thanks for the quick reply.

**Comment by rskelley9** [comment]
Might just come down to the version fo Python one is using. I was using python 3.6 on ubuntu 14.04, so ```
sudo apt-get install python3.6-dev
sudo apt-get install unixodbc-dev
python3.6 -m pip install pyodbc
```
Worked for me.

**Comment by sumitkarn** [comment]
worked for me on ubuntu 14.04. thanks

**Comment by arihantsoganijain** [comment]
@ffedoroff thanks bro, it worked

**Comment by yhhzsd** [comment]
> > src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
> > #include <sql.h>
> > ^
> 
> Your machine does not have the ODBC header files installed. You need to do
> 
> ```
> sudo apt-get install unixodbc-dev
> ```
> 
> and then try installing pyodbc again.

Perfectly solved my problem.

**Comment by karlkaebnick** [comment]
on ubuntu 18.04 and python3.7, python3.7 dev was required for me:

`sudo apt-get install python3.7-dev`

**Comment by Sameer-Keluskar** [comment]
On ubuntu 16.04 and using python3.6 I am still getting the error.
But while installing python3.6-dev  using the following command
sudo apt-get install python3.6-dev
I am getting an error 

> E: Unable to locate package python3.6-dev
> E: Couldn't find any package by glob 'python3.6-dev'
> E: Couldn't find any package by regex 'python3.6-dev'

After doing research on it, I found that we could the following repo:
`sudo add sudo add-apt-repository ppa:deadsnakes/ppa -y && sudo apt-get update` 

But does adding this repo trustworthy?

**Comment by gordthompson** [comment]
deadsnakes has been around for a long time and I've never seen anyone suggest that it is not trustworthy. However, the ultimate decision of whether or not to trust a third-party repo is up to you.

**Comment by evandropomatti** [comment]
This worked for me:

```sh
sudo apt-get install python3.8-dev
```

**Comment by durgaamar** [comment]
Hi,
I'm trying to build a docker image using DockerDesktop <Window 10 OS> of my Python3.7 project. 
I created a Dockerfile and requirements.txt file. Trying to build the docker image but end-up with pyodbc issue as below:
Please suggest what all changes required in my dockerfile to resolve the issue. 

  Building wheel for pyodbc (setup.py): started
  Building wheel for pyodbc (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-v4_phc4e/pyodbc/setup.py'"'"'; __file__='"'"'/tmp/pip-install-v4_phc4e/pyodbc/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-o1oir3_4
       cwd: /tmp/pip-install-v4_phc4e/pyodbc/
  Complete output (14 lines):
  running bdist_wheel
  running build
  running build_ext
  building 'pyodbc' extension
  creating build
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/src
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYODBC_VERSION=4.0.30 -I/usr/local/include/python3.7m -c src/buffer.cpp -o build/temp.linux-x86_64-3.7/src/buffer.o -Wno-write-strings
  In file included from src/buffer.cpp:12:
  src/pyodbc.h:56:10: fatal error: sql.h: No such file or directory
   #include <sql.h>
            ^~~~~~~
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pyodbc
  Running setup.py clean for pyodbc
  Building wheel for visitor (setup.py): started
  Building wheel for visitor (setup.py): finished with status 'done'

**Comment by v-chojas** [comment]
That looks like Linux, not Windows; and the solution is to install `unixodbc-dev`, which has the header files for unixODBC.

**Comment by konidala** [comment]
> I'm see the same issue and both python3-dev and unixodbc-dev are installed.
> I've looked elsewhere and installed every recommended package any other thoughts on how to fix this?
> 
> Python 3.6.4
> Ubuntu 16.04.2 LTS
> 
> ```
>     cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
>     In file included from src/row.cpp:12:0:
>     src/pyodbc.h:45:20: fatal error: Python.h: No such file or directory
>     compilation terminated.
>     error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
> ```

This issue is happening when we try to install pyodbc in virtual environment.

**Comment by ronak2303** [comment]
i am getting same error. i have installed all 
`
yum install -y python-devel
yum install -y unixodbc-devel
yum install -y gcc
yum install -y gcc.c++
`
environment.
python2.6 with redhat 7.7
installed python3 recently as pip was not avaiable.

when i install pyodbc i used pip3 , as pip is not available on my machine.
pip3 install pyodbc

but still error as below on python.h ...

any help here.

> 
  Failed building wheel for pyodbc
  Running setup.py clean for pyodbc
Failed to build pyodbc
Installing collected packages: pyodbc
  Running setup.py install for pyodbc ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-3j7ta5c5/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-ya39qg9u-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_ext
    building 'pyodbc' extension
    creating build
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DPYODBC_VERSION=4.0.30 -I/usr/include/python3.6m -c src/buffer.cpp -o build/temp.linux-x86_64-3.6/src/buffer.o -Wno-write-strings -DHAVE_UNISTD_H -DHAVE_PWD_H -DHAVE_SYS_TYPES_H -DHAVE_LONG_LONG -DSIZEOF_LONG_INT=8 -I/usr/include
    In file included from src/buffer.cpp:12:0:
    src/pyodbc.h:45:20: fatal error: Python.h: No such file or directory
     #include <Python.h>
                        ^
    compilation terminated.
    error: command 'gcc' failed with exit status 1

**Comment by zombozo** [comment]
> > src / pyodbc.h: 56: 17: error fatal: sql.h: No existe tal archivo o directorio 
> > #include <sql.h> 
> > ^
> 
> Su máquina no tiene instalados los archivos de encabezado ODBC. Necesitas hacer
> 
> ```
> sudo apt-get install unixodbc-dev
> ```
> 
> y luego intente instalar pyodbc nuevamente.

un crack, esto me soluciono, todo empezó al utilizar mal un comando remove, tengan cuidado! ahahah

**Comment by 4revalo** [comment]
You have to update your python repository by installing **_yum install python3-devel_**

**Comment by hol-workshop** [comment]
bloody hell, this was required yum install python3-devel
after that pip install pyodbc 
finally 4revalo

**Comment by akristic** [comment]
This helped me from doccuments site:
https://github.com/mkleehammer/pyodbc/wiki/Install
Ubuntu 18.04
On Ubuntu systems, all you need to do is run

sudo apt install python3-pip
sudo apt install unixodbc-dev
sudo apt install python3-dev
pip3 install --user pyodbc

I just didn't use --user because of ERROR: Can not perform a '--user' install. User site-packages are not visible in this virtualenv.

**Comment by kaarthik108** [comment]
> > src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
> > #include <sql.h>
> > ^
> 
> Your machine does not have the ODBC header files installed. You need to do
> 
> ```
> sudo apt-get install unixodbc-dev
> ```
> 
> and then try installing pyodbc again.

I solved it by using this command and then install pyodbc

! apt install unixodbc-dev -y

**Comment by iammathh** [comment]
# Install pyodbc on Centos / RHEL



**Why PyODBC is needed?**

To connect to any Database via Python. (There are specific drivers available for each database, pyodbc serves as a common library)

These are must have libraries for any development work, so its better to have them installed as part of any CentOS / RHEL setup.

```bash
$ sudo yum install unixodbc-dev
$ sudo yum install unixODBC unixODBC-devel
$ sudo yum install epel-release$ sudo yum install gcc-c++
```

These are specific to Python ODBC.
_Before using pip3, make sure you have pip installed._
```bash
$ pip3 --version
$ sudo yum install python3-devel
$ sudo pip3 install pyodbc
```

The installation can be tested as follows
```bash
$ python3

>>> import pyodbc
```
If no errors are thrown, then your installation is perfect.

**Comment by Paulo-Lopes-Estevao** [comment]
Ubuntu 20.04
python3.8

```
sudo apt-get install python3-dev
sudo apt-get install unixodbc-dev
pip3 install pyodbc
```

**Comment by v-chojas** [comment]
There is no point in bumping this issue - please read the documentation: https://github.com/mkleehammer/pyodbc/wiki/Install

**Comment by klentabistado** [comment]
> This worked for me:
> 
> ```shell
> sudo apt-get install python3.8-dev
> ```

worked for me

**Comment by Tonelock** [comment]
> > src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
> > #include <sql.h>
> > ^
> 
> Your machine does not have the ODBC header files installed. You need to do
> 
> ```
> sudo apt-get install unixodbc-dev
> ```
> 
> and then try installing pyodbc again.

Had the same problem running `pyodbc` via `poetry` . This helped me, Thanks!

**Comment by esundberg** [comment]
Fix for Ubuntu 20.04 running Python3.8
`
apt-get install build-essential libssl-dev libffi-dev unixodbc python3.8 python3.8-dev unixodbc-dev
`

**Comment by chaowu2009** [comment]
> > src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
> > #include <sql.h>
> > ^
> 
> Your machine does not have the ODBC header files installed. You need to do
> 
> ```
> sudo apt-get install unixodbc-dev
> ```
> 
> and then try installing pyodbc again.

this works. thank you
